File: 2026_GFG_POTD/JANUARY/26-01-2026/Generate Permutations of an array.js
class Solution {
    permuteDist(arr) {
        const res = [];
        
        const dfs = (i) => {
            if (i === arr.length) {
                res.push([...arr]);
                return;
            }
            for (let j = i; j < arr.length; j++) {
                [arr[i], arr[j]] = [arr[j], arr[i]];
                dfs(i + 1);
                [arr[i], arr[j]] = [arr[j], arr[i]];
            }
        };
        
        dfs(0);
        return res;
    }
}

File: 2026_GFG_POTD/JANUARY/26-01-2026/Generate Permutations of an array.py
class Solution:
    def permuteDist(self, arr):
        res = []
        
        def dfs(i):
            if i == len(arr):
                res.append(arr[:])
                return
            for j in range(i, len(arr)):
                arr[i], arr[j] = arr[j], arr[i]
                dfs(i + 1)
                arr[i], arr[j] = arr[j], arr[i]
        
        dfs(0)
        return res